src/wrappers/any.jl

```diff
@@ -188,27 +196,31 @@ function put(buffer::CxxWrap.StdLib.SharedPtr{LocalMemoryBuffer})
     return put(CxxWrap.CxxWrapCore.__cxxwrap_smartptr_cast_to_base(buffer))
 end
 
-function Base.take!(buffer::CxxWrap.CxxWrapCore.SmartPointer{<:Buffer})
+function Base.take!(buffer::Union{CxxWrap.CxxWrapCore.SmartPointer{<:Buffer},
+                                  CxxWrap.StdLib.SharedPtrAllocated{Buffer}})
```

Comment:
I thouuuught there's a supertype you can use for all smart pointers

Actually, I kinda thought that `CxxWrap.CxxWrapCore.SmartPointer` was that supertype:

```julia
julia> subtypes(CxxWrap.CxxWrapCore.SmartPointer)
3-element Vector{Any}:
 CxxWrap.StdLib.SharedPtr
 CxxWrap.StdLib.UniquePtr
 CxxWrap.StdLib.WeakPtr

julia> subtypes.(subtypes(CxxWrap.CxxWrapCore.SmartPointer))
3-element Vector{Vector{Any}}:
 [CxxWrap.StdLib.SharedPtrAllocated, CxxWrap.StdLib.SharedPtrDereferenced]
 [CxxWrap.StdLib.UniquePtrAllocated, CxxWrap.StdLib.UniquePtrDereferenced]
 [CxxWrap.StdLib.WeakPtrAllocated, CxxWrap.StdLib.WeakPtrDereferenced]
```

Reply:
yeah, I think that second signature is redundant:

```julia
julia> CxxWrap.StdLib.SharedPtrAllocated{Int64} <: CxxWrap.CxxWrapCore.SmartPointer{<:Number}
true
```

I'm also pretty sure that `Buffer` is an abstract type generated by CxxWrap...(the concrete one is the "allocated" version)

Reply:
yeah I was getting a MethodError for some reason so just monkey patched this in...I'll look to remove it